### PR TITLE
8282359: Intermittent WebKit build failure on Windows: C1090: PDB API call failed, error code 23

### PR DESCRIPTION
**Pull request by HimaBinduMeda**
In Jenkins Build, the intermediate failure on windows is observed due to crash in mspdbserv.exe, which generates Program database (.pdb) files.
Disabled the pdb file generation in order to resolve the build failure.
Sanity testing looks fine. Verified build on windows, mac and linux.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed (2 reviews required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer), 1 [Author](https://openjdk.org/bylaws#author))

### Issue
 * [JDK-8282359](https://bugs.openjdk.org/browse/JDK-8282359): Intermittent WebKit build failure on Windows: C1090: PDB API call failed, error code 23

### Reviewers
 * [Kevin Rushforth](https://openjdk.org/census#kcr) (@kevinrushforth - **Reviewer**)
 * [Joeri Sykora](https://openjdk.org/census#sykora) (@tiainen - Author)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jfx.git pull/1079/head:pull/1079` \
`$ git checkout pull/1079`

Update a local copy of the PR: \
`$ git checkout pull/1079` \
`$ git pull https://git.openjdk.org/jfx.git pull/1079/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 1079`

View PR using the GUI difftool: \
`$ git pr show -t 1079`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jfx/pull/1079.diff">https://git.openjdk.org/jfx/pull/1079.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jfx/pull/1079#issuecomment-1491483506)

### CODE REVIEW

**Comment by kevinrushforth** [comment]
I'll do some additional testing early next week. I do have one quick request: Rather than removing or commenting out the compiler and linker flags, can you use `if (PORT STREQUAL "Java")` (or  `if (NOT PORT STREQUAL "Java")`) as was done in #1073 ?

**Comment by johanvos** [comment]
> I'll do some additional testing early next week. I do have one quick request: Rather than removing or commenting out the compiler and linker flags, can you use `if (PORT STREQUAL "Java")` (or `if (NOT PORT STREQUAL "Java")`) as was done in #1073 ?

I agree with this remark.
Also, do you have insight in why removing the debug flag "fixes" the problem? Are we sure that there is not something wrong below the surface that becomes visible in case (but not necessarily limited to this case) the debug flags are on?

File: modules/javafx.web/src/main/native/Source/cmake/OptionsMSVC.cmake [hunk 1]
```diff
@@ -110,11 +110,19 @@ add_compile_options(
 )
 
 if (NOT WTF_CPU_X86)
+    if (PORT STREQUAL "Java")
+    # Create pdb files for debugging purposes, also for Release builds
```
Comment (kevinrushforth) [inline modules/javafx.web/src/main/native/Source/cmake/OptionsMSVC.cmake:114]: Now that this is in an "if...Java" block, the comment is misleading. Maybe change it to something like "Suppress creation of pdb files for Release builds" and also add a "FIXME" or "TODO" to re-enable them for debug builds?

**Comment by kevinrushforth** [comment]
> do you have insight in why removing the debug flag "fixes" the problem? Are we sure that there is not something wrong below the surface that becomes visible in case

The problem is triggered by the generation of the PDB files. PDB files are generated either when the `/Zi` flag is added at compile time or when the `/DEBUG` flag is added at link time. This PR removes both flags.

As a follow-up, we should make sure that the PDB files are generated for DebugNative builds (which are currently broken).

**Review by kevinrushforth — APPROVED** [review]
Testing is green. I'll reapprove if you update the comment I suggested.

File: modules/javafx.web/src/main/native/Source/cmake/OptionsMSVC.cmake (hunk 1 above)
Comment (HimaBinduMeda) [reply modules/javafx.web/src/main/native/Source/cmake/OptionsMSVC.cmake]: Done

**Review by kevinrushforth — APPROVED** [review]

**Review by tiainen — APPROVED** [review]
All builds and tests succeeded here as well.

**Pull request closed without merging**